### ISSUE TITLE
Does crypto-js work with react-native?

### ISSUE THREAD

**Issue by RichardNassar1** [brix/crypto-js#303]
Hello,

I tried to use crypto-js with react native and I got an error. Please advise.

**Comment by rajivshah3** [comment]
I think the issue is that `crypto-js` requires `crypto`, which is a built-in Node.js module. It's not supplied with React Native. We used [react-native-crypto](https://github.com/tradle/react-native-crypto) with [rn-nodeify](https://github.com/tradle/rn-nodeify) to get around this. It should allow `crypto-js` to work properly

**Comment by showtan001** [comment]
+1

**Comment by rajivshah3** [comment]
Ah, the `require` is an issue as well. If I have time I'll contribute a PR for this

**Comment by rohithnair1996** [comment]
how to use react-native-crypto

**Comment by ortonomy** [comment]
Use `^3.0.0` not `^4.0.0`, it works in RN

**Comment by csandman** [comment]
In the patch notes for version `4.0.0` they state that this library will no longer be compatible with React Native.  I tried the route mentioned by @rajivshah3, however this added a lot of bloat, so I think using version `3.3.0` is a better option. to install it, enter:

```
npm install crypto-js@3.3.0 --save
```

**Comment by entronad** [comment]
If the issues caused by secure radom of crypto-js 4.0 bother you, such as too large bundle, unavalible in React Native... Maybe you can try [crypto-es](https://www.npmjs.com/package/crypto-es) .

We merged all updates of crypto-js 4.0 except the secure radom.

**Comment by LogiMateenSA** [comment]
Somehow version 3.0.0 does not work for me

Below is code i am using for setting cookie from node server
```
var cookieValue="Mateen Syed"
cookieValue ? res.cookie(cookieName, CryptoJS.AES.encrypt(cookieValue, "Dummy").toString()) : "" 
console.log("Settnig cookie decoded value ", CryptoJS.AES.encrypt(cookieValue, "Dummy").toString())

```
**Below is result from console**
Settnig cookie decoded value  U2FsdGVkX1+ddRjYsooHJS8z1f6qGzELqsHDYAEikAE=


Below is code i am using to get decrypted value of cookie in react client
```
var decryptvalTemp = CryptoJS.AES.decrypt(poppedValue, "Dummy");
    console.log("decryptvalTemp ", decryptvalTemp)
    console.log("React decoded value ", decryptvalTemp.toString(CryptoJS.enc.Utf8))
```
**Below is output from console.** 
![image](https://user-images.githubusercontent.com/44797233/138566177-6799867f-a1f5-4995-a2db-5463a8b81b3c.png)

Can anyone please help me.

**Comment by gantispam** [comment]
@LogiMateenSA 

```js
    const myObjectToEncrypt = {foo: 'bar'};
    const myObjectToEncryptAsString = JSON.stringify(myObjectToEncrypt);
    const myObjectEncryptedAsString = CryptoJS.AES.encrypt(myObjectToEncryptAsString, 'my-private-key').toString();
    const myObjectDecryptedAsString = CryptoJS.AES.decrypt(myObjectEncryptedAsString, 'my-private-key').toString(CryptoJS.enc.Utf8);
    const myObjectDecrypted = JSON.parse(myObjectDecryptedAsString); // return {foo: 'bar'}
```

**Comment by Sahil6458** [comment]
CryptoJS.HmacSHA256() is not working and gives me **TypeError: undefined is not a function** error. i'm using **react-native-crypto-js** package

**Comment by DavidAPears** [comment]
Same here. I've lost the best part of a day on this. Any help greatly appreciated. 

```
import CryptoJS from "react-native-crypto-js";

 const signature = CryptoJS.HmacSHA256(
    `${encodedHeader}.${encodedPayload}`,
     secret
  );
```
  
 Throws the error:

```
 WARN  Possible Unhandled Promise Rejection (id: 0):
TypeError: _reactNativeCryptoJs.default.HmacSHA256 is not a function. (In '_reactNativeCryptoJs.default.HmacSHA256(encodedHeader + "." + encodedPayload, secret)', '_reactNativeCryptoJs.default.HmacSHA256' is undefined)
```

**Comment by gasparnd** [comment]
react-native-crypto is recommended to use? It has not be updated for 5 years ago

**Comment by afifputra** [comment]
is it still can't solved ?

**Comment by tanmayR18** [comment]
> is it still can't solved ?

I don't think so. I am facing the same issue and not getting the correct solution. Sometimes, the error is "Native module not found," while sometimes it is "Invalid array length."  (using react native)

const iv = CryptoJS.lib.WordArray.random(BLOCK_SIZE);
        console.log('here is the iv', iv);
        const cipher = CryptoJS.AES.encrypt(
          dataToEncrypt,
          CryptoJS.enc.Utf8.parse(publicRsaKey),
          {
            iv: iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7,
          },
        );

**Comment by Parvat-R** [comment]
I had the same issue, and tried using it with the module `react-native-get-random-values` like this:
```
import 'react-native-get-random-values';

import AES from "crypto-js/aes";
import Hex from "crypto-js/enc-hex";
import Utf8 from "crypto-js/enc-utf8";
import SHA256 from "crypto-js/sha256";

export const encrypt = (text: string, key: string): string => {
    return AES.encrypt(text, key).toString();
};

export const decrypt = (cipher: string, key: string): string => {
    const bytes = AES.decrypt(cipher, key);
    return bytes.toString(Utf8);
};

export const hash = (text: string): string => {
    return SHA256(text).toString(Hex);
};
```
and it worked!
